### PR TITLE
Bluetooth: Channel Sounding: Add `bt_cs_set_default_settings`

### PR DESCRIPTION
**Pull request by sean-madigan**
Add CS set default settings command for the host, along with kconfig and shell support.

### CODE REVIEW

File: subsys/bluetooth/host/cs.c [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(conn->handle);
+	cp->max_tx_power = params->max_tx_power;
+	cp->cs_sync_antenna_selection = params->cs_sync_antenna_selection;
+	cp->role_enable = 0;
+
+	if (params->enable_initiator_role) {
+		cp->role_enable &= BT_HCI_OP_LE_CS_INITIATOR_ROLE_MASK;
```
Comment (olivier-le-sage) [inline subsys/bluetooth/host/cs.c:40]: shouldn't this be |= ?

File: subsys/bluetooth/host/cs.c [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#include "conn_internal.h"
+
+#if defined(CONFIG_BT_CHANNEL_SOUNDING)
+int bt_cs_set_default_settings(struct bt_conn *conn,
+			       const struct bt_cs_set_default_settings_param *params)
+{
+	struct bt_hci_cp_le_cs_set_default_settings *cp;
+	struct net_buf *buf;
+
+	if (params->max_tx_power > BT_HCI_OP_LE_CS_MAX_MAX_TX_POWER ||
```
Comment (olivier-le-sage) [inline subsys/bluetooth/host/cs.c:23]: I think the controller also has to check this, and returns "invalid command parameters" in this case. So we could choose to leave this out

**Comment by olivier-le-sage** [comment]
I would slightly prefer channel_sounding.c and channel_sounding.h over acronyms.

I think that's a bit nicer for people who are less familiar, or browsing the code casually

Feel free to leave it as is

File: subsys/bluetooth/host/cs.c (hunk 2 above)
Comment (sean-madigan) [reply subsys/bluetooth/host/cs.c:23]: yeah good point, no need to duplicate the logic

File: subsys/bluetooth/host/cs.c (hunk 1 above)
Comment (sean-madigan) [reply subsys/bluetooth/host/cs.c:40]: of course

**Comment by sean-madigan** [comment]
> I would slightly prefer channel_sounding.c and channel_sounding.h over acronyms.
> 
> I think that's a bit nicer for people who are less familiar, or browsing the code casually
> 
> Feel free to leave it as is

I may leave as is, I think cs will be a useful acronym to use

**Review by olivier-le-sage — APPROVED** [review]

File: subsys/bluetooth/shell/cs.c [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(
+	cs_cmds,
+	SHELL_CMD_ARG(
+		set_default_settings, NULL,
+		"<Enable initiator role: true, false> <Enable reflector role: true, false> "
+		" <CS_SYNC antenna selection: 0x01 - 0x04, 0xFE, 0xFF> <Max TX power: -127 - 20>",
+		cmd_set_default_settings, 1, 4),
```
Comment (theob-pro) [inline subsys/bluetooth/shell/cs.c:107]: Shouldn't it be as follow? The arguments are mandatory right?

```suggestion
		cmd_set_default_settings, 5, 0),
```

Comment (sean-madigan) [reply subsys/bluetooth/shell/cs.c:107]: Good point

File: tests/bluetooth/shell/testcase.yaml [hunk 4]
```diff
@@ -43,6 +43,13 @@ tests:
     platform_allow:
       - native_sim
     build_only: true
+  bluetooth.shell.channel_sounding:
+    extra_configs:
+      - CONFIG_BT_CHANNEL_SOUNDING=y
+      - CONFIG_BT_CTLR=n
+    platform_allow:
+      - native_posix
```
Comment (theob-pro) [inline tests/bluetooth/shell/testcase.yaml:51]: Please use `native_sim`. `native_posix` is deprecated: https://docs.zephyrproject.org/latest/boards/native/native_posix/doc/index.html

```suggestion
      - native_sim
```

**Review by theob-pro — CHANGES_REQUESTED** [review]

**Review by theob-pro — DISMISSED** [review]

File: include/zephyr/bluetooth/cs.h [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	BT_CS_ANTENNA_SELECTION_OPT_THREE = BT_HCI_OP_LE_CS_ANTENNA_SEL_THREE,
+	/** Use antenna identifier 4 for CS_SYNC packets. */
+	BT_CS_ANTENNA_SELECTION_OPT_FOUR = BT_HCI_OP_LE_CS_ANTENNA_SEL_FOUR,
+	/** Use antennas in repetitive order from 1 to 4 for CS_SYNC packets. */
+	BT_CS_ANTENNA_SELECTION_OPT_REPETITIVE = BT_HCI_OP_LE_CS_ANTENNA_SEL_REP,
+	/** No recommendation for local controller antenna selection. */
+	BT_CS_ANTENNA_SELECTION_OPT_NO_RECOMMENDATION = BT_HCI_OP_LE_CS_ANTENNA_SEL_NONE,
+};
+
+/** Connection subrating parameters for LE connections */
```
Comment (olivier-le-sage) [inline include/zephyr/bluetooth/cs.h:45]: noticed this now

suggest copying some text from the core spec, like "default CS
settings in the local Controller"

**Review by theob-pro — DISMISSED** [review]

File: subsys/bluetooth/host/cs.c [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	cp->max_tx_power = params->max_tx_power;
+	cp->cs_sync_antenna_selection = params->cs_sync_antenna_selection;
+	cp->role_enable = 0;
+
+	if (params->enable_initiator_role) {
+		cp->role_enable |= BT_HCI_OP_LE_CS_INITIATOR_ROLE_MASK;
+	}
+
+	if (params->enable_reflector_role) {
+		cp->role_enable |= BT_HCI_OP_LE_CS_INITIATOR_ROLE_MASK;
```
Comment (olivier-le-sage) [inline subsys/bluetooth/host/cs.c:39]: should be bit 1, BT_HCI_OP_LE_CS_REFLECTOR_ROLE_MASK

**Review by olivier-le-sage — CHANGES_REQUESTED** [review]

**Review by olivier-le-sage — APPROVED** [review]

**Review by bugo-nordic — APPROVED** [review]

**Review by theob-pro — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**